src/basic/time-util.h

```diff
@@ -29,7 +29,11 @@
 typedef uint64_t usec_t;
 typedef uint64_t nsec_t;
 
-#define PRI_NSEC PRIu64
+#if defined __x86_64__ && defined __ILP32__ && defined __GLIBC__
+#  define PRI_NSEC PRIi64
+#else
+#  define PRI_NSEC "ld"
+#endif
```

Comment:
nah, this is not OK, nsec_t is defined by systemd and not any libc. and PRI_NSEC should map to our type which is unconditionally uint64_t and should stay that way.

Please, let's just use PRI_NSEC the way it is right now, and just cast the offending variable to nsec_t explicitly while printing it to avoid such confusion